src/puppetlabs/ezbake/core.clj

```diff
@@ -499,25 +510,92 @@ Dependency tree:
+      {:project (:name lein-project)
+       :summary (:description lein-project)
+       :description (format "%s (%s)"
+                            (:description lein-project)
+                            (deputils/generate-manifest-string lein-project))
+       :uberjar-name (:uberjar-name lein-project)
+       :additional-uberjars (mapv (fn [filename] {:uberjar filename}) additional-uberjars)
+       :is-pe-build (format "%s" (= (get-local-ezbake-var lein-project :build-type "foss") "pe"))
+       :repo-name (format "%s" (get-local-ezbake-var lein-project :repo-target ""))})))
+
+(schema/defn get-additional-uberjars
+  [lein-project]
+  (when-let [dependencies-vector (get-in lein-project [:lein-ezbake :additional-uberjars])]
+    dependencies-vector))
+
+(schema/defn resolve-dependency! :- File
+  "Resolves a single dependency and returns a File pointing to it's jar"
+  [[project-symbol version :as dependency-coordinates]]
+  (lein-main/info "Resolving dependency for " dependency-coordinates)
+  (let [project {:dependencies [dependency-coordinates]
+                 :repositories [["releases" {:url "https://clojars.org/repo"}]]}
+        ; resolved-dependencies will be a list of jars, one for each dependency
+        ; in the project, including the jar for this project
+        resolved-dependencies (lein-classpath/resolve-managed-dependencies :dependencies nil project)
+        jar-regex (re-pattern (format "%s-%s.jar" (name project-symbol) version))
+        jar-file (first (filter #(re-find jar-regex (.getName %))
+                                resolved-dependencies))]
+    jar-file))
+
+(schema/defn extract-project-from-jar! :- schema/Str
+  "Extracts the project.clj file out of the given jar into the destination directory"
+  [jar :- File
+   destination-dir :- schema/Str]
+  (fs/mkdirs destination-dir)
+  (let [jar-file (JarFile. jar)
+        jar-entry (.getJarEntry jar-file "project.clj")
+        project-string (-> jar-file
+                           ; Get an input string for project.clj
+                           (.getInputStream jar-entry)
+                           slurp)
+        out-file-path (str destination-dir "/project.clj")]
+    (spit out-file-path project-string)
+    out-file-path))
+
+(schema/defn build-uberjar-from-coordinates! :- File
+  "Build an uberjar from maven coordinates and return a file pointing to the built uberjar"
+  [[project-symbol version :as dependency-coordinates]]
+  (let [dependency-jar (resolve-dependency! dependency-coordinates)
+        destination-dir (format "%s/%s-%s"
+                                additional-uberjar-checkouts-dir
+                                (name project-symbol)
+                                version)
+        project-file (extract-project-from-jar! dependency-jar destination-dir)
+        project-map (-> project-file
+                        project/read
+                        (update :dependencies conj dependency-coordinates))]
```

Comment:
For future maintainer, might be good to add a comment here about why this is necessary.  It does make sense to me, though, based on our prior conversations about it.